File: iPhone/VerBank-CSTSv3-IOSClientApi/ClientStation/api/info/InfoOperator_TRADESERV1014.m
//
//  InfoOperator_TRADESERV1014.m
//  VerBank-IOSClientStation
//
//  Created by Allone on 15/8/31.
//  Copyright (c) 2015年 zhanglei. All rights reserved.
//

#import "InfoOperator_TRADESERV1014.h"

#import "Info_TRADESERV1014.h"
#import "MessageToAccount.h"
#import "MTP4CommDataInterface.h"
#import "API_IDEventCaptain.h"

@implementation InfoOperator_TRADESERV1014

- (void)onInfo:(InfoFather *)info {
    Info_TRADESERV1014 *tinfo = (Info_TRADESERV1014 *)info;
    MessageToAccount *message = [tinfo getMessage];
    [message setIsRead:FALSE];
    
    [API_IDEventCaptain fireEventChanged:DATA_ON_MessageReceive eventData:message];
}

@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/client/station/api/info/InfoOperator_TRADESERV1014.java
package jedi.ex01.client.station.api.info;

import jedi.ex01.CSTS3.comm.info.InfoFather;
import jedi.ex01.CSTS3.comm.info.Info_TRADESERV1014;
import jedi.ex01.CSTS3.comm.struct.MessageToAccount;
import jedi.ex01.CSTS3.proxy.comm.MTP4CommDataInterface;
import jedi.ex01.client.station.api.event.API_IDEventCaptain;
import jedi.ex01.client.station.api.event.API_IDEvent_NameInterface;

public class InfoOperator_TRADESERV1014 extends InfoOperator {

	@Override
	public void onInfo(InfoFather _info) {
		Info_TRADESERV1014 info = (Info_TRADESERV1014) _info;
		MessageToAccount message = info.getMessage();
		message.setIsRead(MTP4CommDataInterface.FALSE);
		API_IDEventCaptain.fireEventChanged(API_IDEvent_NameInterface.DATA_ON_MessageReceive, info
				.getMessage());
	}

}
